backstopper-core/src/test/java/com/nike/backstopper/apierror/ApiErrorBaseTest.java

```diff
@@ -2,27 +2,34 @@
 
 import com.nike.internal.util.MapBuilder;
 
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 
 import java.util.Map;
 import java.util.UUID;
 
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.CoreMatchers.notNullValue;
-import static org.junit.Assert.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /**
  * Tests the functionality of {@link ApiErrorBase}
  */
+@RunWith(DataProviderRunner.class)
 public class ApiErrorBaseTest {
 
     @Test(expected = IllegalArgumentException.class)
-    public void constructorShouldThrowIllegalArgumentExceptionIfPassedNullName() {
+    public void constructor_should_throw_IllegalArgumentException_null_name() {
         new ApiErrorBase(null, 42, "some error", 400);
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void constructor_should_throw_IllegalArgumentException_null_error_code() {
+        new ApiErrorBase(null, null, "some error", 400, null);
```

Comment:
The `IllegalArgumentException` being thrown here is likely due to the null name, not the null error code you're trying to test. I'd recommend switching from `@Test(expected = ...)` to AssertJ's `catchThrowable` and verify the message so you know the exception was thrown for the desired reason.

Reply:
i've changed this test and the other in the class to use `catchThrowable`